akka-projection-cassandra/src/main/scala/akka/projection/cassandra/internal/CassandraProjectionImpl.scala

```diff
@@ -77,13 +79,45 @@ import akka.stream.scaladsl.Source
     settingsOpt: Option[ProjectionSettings],
     handler: Handler[Envelope])
     extends javadsl.CassandraProjection[Envelope]
-    with scaladsl.CassandraProjection[Envelope] {
+    with scaladsl.CassandraProjection[Envelope]
+    with javadsl.AtLeastOnceCassandraProjection[Envelope]
+    with scaladsl.AtLeastOnceCassandraProjection[Envelope] {
   import CassandraProjectionImpl._
   import HandlerRecoveryImpl.applyUserRecovery
 
-  override def withSettings(settings: ProjectionSettings): CassandraProjectionImpl[Offset, Envelope] = {
+  override def withSettings(settings: ProjectionSettings): CassandraProjectionImpl[Offset, Envelope] =
     new CassandraProjectionImpl(projectionId, sourceProvider, strategy, Option(settings), handler)
-  }
+
+  /**
+   * Settings for AtLeastOnceCassandraProjection
+   */
+  override def withSaveOffsetAfterEnvelopes(afterEnvelopes: Int): CassandraProjectionImpl[Offset, Envelope] =
```

Comment:
I think we should have both parameters together, because if changing one the user should think about the other
```
def withSaveOffset(afterEnvelopes: Int, orAfterDuration: FiniteDuration)
```

Reply:
Ok